tests/test_aot.rs

```diff
@@ -375,3 +375,48 @@ pub fn test_aot_rvc_pageend() {
     assert!(result.is_ok());
     assert_eq!(result.unwrap(), 0);
 }
+
+pub struct OutOfCyclesSyscall {}
+
+impl<Mac: SupportMachine> Syscalls<Mac> for OutOfCyclesSyscall {
+    fn initialize(&mut self, _machine: &mut Mac) -> Result<(), Error> {
+        Ok(())
+    }
+
+    fn ecall(&mut self, machine: &mut Mac) -> Result<bool, Error> {
+        let code = &machine.registers()[A7];
+        if code.to_i32() != 1111 {
+            return Ok(false);
+        }
+        machine.add_cycles_no_checking(100)?;
+        let result = machine.registers()[A0]
+            .overflowing_add(&machine.registers()[A1])
+            .overflowing_add(&machine.registers()[A2])
+            .overflowing_add(&machine.registers()[A3])
+            .overflowing_add(&machine.registers()[A4])
+            .overflowing_add(&machine.registers()[A5]);
+        machine.set_register(A0, result);
+        Ok(true)
+    }
+}
+
+#[test]
+pub fn test_aot_outofcycles_in_syscall() {
+    let buffer = fs::read("tests/programs/syscall64").unwrap().into();
+    let mut aot_machine =
+        AotCompilingMachine::load(&buffer, Some(Box::new(|_| 1)), ISA_IMC, VERSION0).unwrap();
+    let code = aot_machine.compile().unwrap();
+    let asm_core = AsmCoreMachine::new(ISA_IMC, VERSION0, 20);
+    let core = DefaultMachineBuilder::new(asm_core)
+        .instruction_cycle_func(Box::new(|_| 1))
+        .syscall(Box::new(OutOfCyclesSyscall {}))
+        .build();
+    let mut machine = AsmMachine::new(core, Some(&code));
+    machine
+        .load_program(&buffer, &vec!["syscall".into()])
+        .unwrap();
+    let result = machine.run();
+    assert!(result.is_err());
+    assert_eq!(result.unwrap_err(), Error::InvalidCycles);
+    assert_eq!(machine.machine.cycles(), 108);
```

Comment:
We should add an assertion line here, ensuring `A0` is indeed overwritten, meaning the syscall completes execution despite the cycle error